File: quote/Projects/PluginModel/Shipping.vb
Imports System.Collections.ObjectModel

Public Class Shipping

    Public Shared Property Dictionary As New SortedDictionary(Of String, Decimal)

    Public Shared Property Descriptions As New ReadOnlyCollection(Of String)( {""} )

    Public Shared Function Lookup(ByVal Description As String) As Decimal
    
        If (_Dictionary.ContainsKey(Description)) Then
            Return _Dictionary(Description)
        End If
        
        Return 0
        
    End Function

End Class

File: quote/Projects/PluginModel/Shipping.cs
namespace Model
{
    using System;
    using System.Collections.Generic;
    using System.Collections.ObjectModel;
    using System.Linq;

    public sealed class Shipping
    {
        private static SortedDictionary<string, decimal> dictionary = new SortedDictionary<string, decimal>();

        private static ReadOnlyCollection<string> descriptions = new ReadOnlyCollection<string>(new string[] { string.Empty });

        private Shipping()
        {
        }

        public static SortedDictionary<string, decimal> Dictionary 
        {
            get { return dictionary; }
        }

        public static ReadOnlyCollection<string> Descriptions 
        {
            get { return descriptions; }
        }

        public static void SetupDescriptions(ReadOnlyCollection<string> descriptions)
        {
            Shipping.descriptions = descriptions;
        }

        public static decimal Lookup(string description)
        {
            if (dictionary.ContainsKey(description))
            {
                return dictionary[description];
            }

            return 0;
        }
    }
}
